_spinnaker/terraform_integration.md

```diff
@@ -201,215 +227,54 @@ After you configure your Git repository and Gate proxy access, perform the follo
-  },
-  "terraformVersion": "0.12.1",
-  "type": "terraform",
-  "workspace": "foo"
-}
-```
-
-The apply stage requires this (or similar) SpEL expression(s) to pass the data in.
+**Note**: If you used the Terraform integration by editing the JSON representation of the stage, stages are automatically converted to the UI. For a tour of the UI, see the [Terraform Integration UI video](https://www.youtube.com/watch?v=Xsjql3g-wtU)
+
+![Terraform Stage in Deck](/images/terraform_stage_ui.png)
+
+The Terraform Integration exposes a new stage in Spinnaker called **Terraform**. To use the stage, perform the following steps:
+
+1. In Deck, select the Application and pipeline you want to add the Terraform Integration stage to.
+2. Configure the pipeline and add a stage.
+3. For **Type**, select **Terraform**.
+4. Add a **Stage Name**.
+5. Configure the Terraform Integration stage.
+    The available fields may vary slightly depending on what you configure for the stage: 
+    * **Basic Settings**
+      * **Terraform Version**:  Terraform version to use. **SYSTEM_DEFINED** refers to the Terraform version set in `terraformer-local.yml`. All Terraform stages within a pipeline that modify state (apply, output, destroy) must use the same version.
+      * **Action**: Terraform action to perform. You can select any of the following actions:
+        * **Plan**: The output of the plan command is saved to a base64-encoded Spinnaker artifact and is injected into context.  You can use this artifact with a webhook to send the plan data to an external system or to use it in an `apply` stage. Optionally, you can select **Plan for Destroy** to view what Terraform destroys if you run the Destroy action.
+        * **Apply**: Run `terraform apply`. Optionally, you can ignore state locking. Armory recommends you do not ignore state locking because it can lead to state corruption. Only use this setting if you understand the consequences. 
+        * **Destroy**: Run `terraform destroy`. Optionally, you can ignore state locking. Armory recommends you do not ignore state locking because it can lead to state corruption. Only use this setting if you understand the consequences.
+        * **Output**: Run `terraform output`. 
+      * **Profile**: Select a profile to use when executing Terraform. Variables and conditions provided to the selected profile get injected into the Terraform runtime for the lifetime of the execution. These profiles are defined in ``.hal/default/profiles/terraformer-local.yml`.
+      * **Targets**: Scope execution to a certain subset of resources.
+      * **Workspace**: [Terraform workspace](https://www.terraform.io/docs/state/workspaces.html) to use. The workspace gets created if it doesn't already exist.
+    * **Main Terraform Artifact**
+      * **Expected Artifact**: Required for **Plan** stages. Select or define only one `git/repo` artifact. Currently, GitHub and BitBucket artifacts are supported. 
```

Comment:
Isn't this required for all commands, not just plan? I also don't think it needs to be a git/repo artifact, but that's probably the most common.

Reply:
That line was in the original docs, so let's go with your understanding!